### ISSUE TITLE
MD033 has warning when used as documented

### ISSUE THREAD

**Issue by skyzyx** [rvben/rumdl#627]
### rumdl version

0.1.91

### Installation method

mise

### Bug description

When porting rules from Markdownlint (David Anson's) to rumdl, MD033 fails to allow HTML elements which are explicitly allow-listed in the configuration. Instead, rumdl triggers a warning.

### Steps to reproduce

1. Create a README.md with the following contents:

    ```markdown
    <div><img src="docs/auth@2x.png" alt="Match the authorization code" width="50%"></div>
    ```

    (Since `width=50%` has no Commonmark-standardized syntax, falling back to HTML is the only reliably-portable solution. HTML fallback is also explicitly part of Markdown's design per John Gruber's 2004 documentation. [citation needed])

2. Create a `.rumdl.toml` definition with the following contents:

    ```toml
    [rules]
      # https://rumdl.dev/md033/
      [rules.MD033]
        # no-inline-html
        enabled = true
        allowed-elements = [
          "a",
          "b",
          "br",
          "code",
          "details",
          "div",
          "img",
          "li",
          "nobr",
          "ol",
          "p",
          "pre",
          "span",
          "summary",
          "ul",
        ]
        br-style = "backslash"
        drop-attributes = [
          "class",
          "id",
          "rel",
          "style",
          "target",
        ]
        fix = true
        fix-mode = "relaxed"
        strip-wrapper-elements = [
          "p",
        ]
        table-allowed-elements = []
    ```

3. Run `rumdl`:

    ```bash
    rumdl check --fix --statistics --output-format grouped
    ```

### Expected behavior

No warnings, since these elements were specifically allowed in the config.

### Actual behavior

```text
README.md:
  MD033:
    X:Y Inline HTML found: <div>
    X:Y Inline HTML found: <img src="docs/auth@2x.png" alt="Match the authorization code" width="50%">

Issues: Found 2 issues in 1/1 files (Zms)

Rule Violation Statistics:
Rule     Violations   Fixable  Percentage
--------------------------------------------------
MD033    2            -         100.0%
--------------------------------------------------
Total    2            0         100.0%
```

### Configuration

```toml
[rules]
  # https://rumdl.dev/md033/
  [rules.MD033]
    # no-inline-html
    enabled = true
    allowed-elements = [
      "a",
      "b",
      "br",
      "code",
      "details",
      "div",
      "img",
      "li",
      "nobr",
      "ol",
      "p",
      "pre",
      "span",
      "summary",
      "ul",
    ]
    br-style = "backslash"
    drop-attributes = [
      "class",
      "id",
      "rel",
      "style",
      "target",
    ]
    fix = true
    fix-mode = "relaxed"
    strip-wrapper-elements = [
      "p",
    ]
    table-allowed-elements = []
```

### Operating system

macOS

### Additional context

Old (working) config for Markdownlint (including `markdownlint-cli`).

https://github.com/northwood-labs/.github/blob/f8437768da0b1b8da7fd61e84debb0991993a057/.markdownlint.json

**Comment by rvben** [comment]
This looks resolved since **v0.1.92**. With your config (`MD033` enabled, `allowed-elements` incl. `div`/`img`, `fix-mode="relaxed"`) and `<div><img ...></div>`, current rumdl (v0.2.0) reports zero MD033 warnings.

Could you upgrade (`v0.2.0` is out) and confirm? I'll close once you've verified.

**Comment by skyzyx** [comment]
Confirmed as working. Thank you.
